controller/volume_controller.go

```diff
@@ -1773,30 +1773,6 @@ func (c *VolumeController) closeVolumeDependentResources(v *longhorn.Volume, e *
 		}
 	}
 
-	// check if any replica has been RW yet
```

Comment:
1. This function will be invoked when the volume will be updated to `detaching` or `detached`. Does this make sense?
2. Even if we ignore the context, the function name `closeXXXX` means we have to try our best to stop all volume-related resources including rebuilding replicas. Leaving such kind of rebuilding replica is not expected in terms of callers' perspective.

Reply:
We still stop the rebuilding replica on 1816. The difference here is that we do so AFTER the engine stops, at the same time we stop the other replicas.

I am not 100% sure why we had logic to stop the rebuilding replica sooner. I assume it is because we thought that it was safe to do so (since it cannot lead to corruption like the comment on 1789 indicates). However, my analysis indicates that it can still cause this other issue, so IMO moving it makes sense.

WDYT?

Reply:
OK. Got it. Stopping replicas including rebuilding ones **after the engine stop** makes sense and it's supposed to be. In this perspective, we cannot stop rebuilding replicas first during detachment or cleanup.